### PR TITLE
Issue #44, added github api

### LINKED ISSUE #44: [API Request] Call GitHub APIs to get a list of all the current open issues for org
For all the repositories in the [organisation Project Travel Mate](https://github.com/project-travel-mate), make a call to [List issues for a repository GitHub API](https://developer.github.com/v3/issues/#list-issues-for-a-repository) and fetch all the currently open issues. Include following fields for each object.
+ Issue title
+ Created at
+ Number of comments 
+ Repository URL 
+ Issue number
+ Tags

### PR DESCRIPTION
**Pull request by techac**
# Description

Added GitHub API  for listing issues of a repo

Fixes #44 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `flake8`
- [x] `python manage.py test`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: api/modules/github/views.py [hunk 1]
```diff
@@ -45,3 +45,43 @@ def get_contributors(request, project):
...
+
+        if api_response.status_code == 401:
+            raise Exception("Authentication fails. Invalid github access token.")
+        response = []
+        for issue in api_response_json:
+            labels_length = len(issue['labels'])
+            tags = []
+            # Making custom dictionary for tags
+            for i in range(0, labels_length):
+                mydict = {k: v for k, v in issue["labels"][i].items() if k in ["name"]}
```
Comment (prabhakar267) [inline api/modules/github/views.py]: change the name to something more understandable

File: api/modules/github/views.py [hunk 2]
```diff
@@ -45,3 +45,43 @@ def get_contributors(request, project):
...
+                tags.append(mydict)
+            result = IssueResponse(
+                title=issue['title'],
+                created_at=issue['created_at'],
+                comments=issue['comments'],
+                issue_number=issue['number'],
+                repository_url=issue['repository_url'],
+                labels=tags
```
Comment (prabhakar267) [inline api/modules/github/views.py:85]: I'm getting labels like 
```
        "tags": [
            {
                "name": "gssoc"
            },
            {
                "name": "gssoc-pro"
            },
            {
                "name": "help wanted"
            }
        ]
```
Please change it to a list of strings

File: api/modules/github/views.py [hunk 3]
```diff
@@ -45,3 +45,43 @@ def get_contributors(request, project):
...
+def get_issues(request, project):
+    """
+        Return list of issues
+        :param request:
+        :param project:
+        :return: 503 if github api fails
+        :return: 200 successful
+        """
+    try:
+        api_response = requests.get(GITHUB_API_GET_ISSUES_URL.format(project_name=project))
```
Comment (prabhakar267) [inline api/modules/github/views.py]: If I give an incorrect project name, I'm getting `503` with error message `"string indices must be integers"`
Add a better error message with `404` status code

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

**Comment by techac** [comment]
@prabhakar267 you may review the changes

**Review by prabhakar267 — APPROVED** [review]

**Pull request merged**